src/cli/NativeMsgManager.cpp

```diff
@@ -0,0 +1,115 @@
+#include "NativeMsgManager.h"
+#include <QCoreApplication>
+#include <QDebug>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QMap>
+#include <QJsonArray>
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QList>
+#include <QPair>
+#include <QRegularExpression>
+#include <QSet>
+#include <QSettings>
+#include <QStandardPaths>
+#include <QStringList>
+
+
+static QRegularExpression firefoxExtensionPattern("^\\{[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\\}$|^[a-z0-9-._]*@[a-z0-9-._]+$", QRegularExpression::CaseInsensitiveOption);
```

Comment:
WHy not one for Chromium? Could you add a comment explaining this regex.

Reply:
I'll add a comment. It's described a little later on as well that Firefox only allows "valid" extension ids in the manifest file otherwise it will fail to be parsed.